src/gradescopeapi/api/api.py

```diff
@@ -1,200 +1,359 @@
-              otherwise returns False
+        dict: A dictionary containing the extensions, where the keys are user IDs and the values are Extension objects.
+
+    Raises:
+        HTTPException: If the request to get extensions fails, with a 500 Internal Server Error status code and the error message.
     """
-    return False
+    try:
+        extensions = get_extensions(
+            session=session,
+            course_id=extension_data.course_id,
+            assignment_id=extension_data.assignment_id,
+        )
+        return extensions
+    except RuntimeError as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@app.post("/assignments/extensions/update")
+def update_extension(
+    update_extension_data: UpdateExtensionData,
+    session: requests.Session = Depends(get_gs_connection_session),
+):
+    """
+    Update the extension for a student on an assignment.
+
+    Args:
+        update_extension_data (UpdateExtensionData): Pydantic model containing the course_id, assignment_id, user_id,
+            and optional release_date, due_date, and late_due_date.
+        session (requests.Session, optional): The session object used for making HTTP requests.
+            Defaults to the session instance provided by the get_gs_connection_session dependency.
 
+    Returns:
+        dict: A dictionary with a "message" key indicating if the extension was updated successfully.
 
-def update_student_release_date(
-    token: TokenModel, student_id: str, course_id: str, assignment_id: str
-) -> bool:
-    """_summary_
+    Raises:
+        HTTPException: If the extension update fails, with a 400 Bad Request status code and the error message.
+        HTTPException: If a ValueError is raised (e.g., invalid date order), with a 400 Bad Request status code and the error message.
+        HTTPException: If any other exception occurs, with a 500 Internal Server Error status code and the error message.
+    """
+    try:
+        success = update_student_extension(
+            session=session,
+            course_id=update_extension_data.course_id,
+            assignment_id=update_extension_data.assignment_id,
+            user_id=update_extension_data.user_id,
+            release_date=update_extension_data.release_date,
+            due_date=update_extension_data.due_date,
+            late_due_date=update_extension_data.late_due_date,
+        )
+        if success:
+            return {
+                "message": "Extension updated successfully",
+                "status_code": status.HTTP_200_OK,
+            }
+        else:
+            raise HTTPException(status_code=400, detail="Failed to update extension")
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@app.post("/assignments/upload")
+async def upload_assignment_files(
```

Comment:
Is there a reason this is async but none of the other functions are? I presume it is so that it doesn't block on uploading the files which might take a while but I don't know if this makes sense for an API or if that is what this function does. When the user sends an HTTP request they won't get 2 responses, unless they do?